src/command/mcl.zig

```diff
@@ -1896,6 +1959,186 @@ fn mclWaitRecoverSlider(params: [][]const u8) !void {
+    var buf_len = prefix.len;
+
+    var register_iterator = log.register_lists.iterator();
+    while (register_iterator.next()) |reg_entry| {
+        if (!log.register_lists.get(reg_entry.key)) continue;
+        @memcpy(info_buffer[buf_len .. buf_len + @tagName(reg_entry.key).len], @tagName(reg_entry.key));
+        @memcpy(info_buffer[buf_len + @tagName(reg_entry.key).len .. buf_len + @tagName(reg_entry.key).len + 1], ",");
+        buf_len += @tagName(reg_entry.key).len + 1;
+    }
+    std.log.info("{s}", .{info_buffer[0 .. buf_len - 1]});
+
+    const path = params[3];
+    var path_buffer: [512]u8 = undefined;
+    const file_path = if (path.len > 0) path else p: {
+        var timestamp: u64 = @intCast(std.time.timestamp());
+        timestamp += std.time.s_per_hour * 9;
+        const days_since_epoch: i32 = @intCast(timestamp / std.time.s_per_day);
+        const ymd =
+            command.chrono.date.YearMonthDay.fromDaysSinceUnixEpoch(days_since_epoch);
+        const time_day: u32 = @intCast(timestamp % std.time.s_per_day);
+        const time = try command.chrono.Time.fromNumSecondsFromMidnight(time_day, 0);
+
+        break :p try std.fmt.bufPrint(
+            &path_buffer,
+            "mmc-register-{s}-{s}-{}.{:0>2}.{:0>2}-{:0>2}.{:0>2}.{:0>2}.csv",
+            .{
+                line_name,
+                info_buffer[prefix.len .. buf_len - 1],
+                ymd.year,
+                ymd.month.number(),
+                ymd.day,
+                time.hour(),
+                time.minute(),
+                time.second(),
+            },
+        );
+    };
+    std.log.info("{s}", .{file_path});
+
+    log.log_file = try std.fs.cwd().createFile(file_path, .{});
+    if (log.log_file) |f| {
+        register_iterator.index = 0;
+        while (register_iterator.next()) |reg_entry| {
+            if (!log.register_lists.get(reg_entry.key)) continue;
+            for (log.station_indices) |station_idx| {
+                inline for (@typeInfo(@TypeOf(reg_entry.key)).@"enum".fields) |register_enum| {
+                    if (@intFromEnum(reg_entry.key) == register_enum.value) {
+                        var _buffer: [64]u8 = undefined;
+                        try registerFieldToString(
+                            f.writer(),
+                            try std.fmt.bufPrint(&_buffer, "station{d}", .{station_idx + 1}),
+                            "",
+                            @FieldType(RegisterLogging.Registers, register_enum.name),
+                        );
+                        break;
+                    }
+                }
+            }
+        }
+        try f.writer().writeByte('\n');
+    }
+    log_line[line_idx] = log;
+}
+
+/// Write the register values to the logging file specified by
+/// line_name parameter. If the line has not ben set for logging,
+/// it will return error.
+fn logRegisters(params: [][]const u8) !void {
+    const line_name: []const u8 = params[0];
```

Comment:
Is there a need for this command to accept a line parameter? The selected lines to log cannot change anyway after the `SET_...` command.

Reply:
I think this is needed since each line has different logging file in the current code. It might be useful if we have multiple lines running and trigger logging of specific line after a command has been passed to that line. In addition, it will be easier to analyze the behavior of each line for the engineers if we separate the logging file.

Reply:
Let's discuss more during today's meeting!